p2p/queue.go

```diff
@@ -26,17 +28,14 @@ type queue interface {
 // fifoQueue is a simple unbuffered lossless queue that passes messages through
 // in the order they were received, and blocks until message is received.
 type fifoQueue struct {
```

Comment:
Shall we move this to its own file so that it is in keeping with the other queue implementations or perhaps a better question is: Is this queue implementation temporary?

Reply:
This queue is temporary :)